### PR TITLE
Blog posts/how to experiment rapidly without losing rigour

### PR DESCRIPTION
**Pull request by MahanaC**
A data science blog post on sequential experiment design, suggested audience is data scientists and related professionals with experience in frequentist experiment design. Feedback very welcome!

Written by @harrysalmon and myself.

### CODE REVIEW

File: _posts/2018-10-12-how-to-experiment-rapidly-without-losing-rigour.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+[^group-sequential]: "Interim Monitoring of Group Sequential Trials Using Spending Functions for the Type I and Type II Error Probabilities" [http://journals.sagepub.com/doi/abs/10.1177/009286150103500408](http://journals.sagepub.com/doi/abs/10.1177/009286150103500408). Accessed 5 Sep. 2017.
+[^interim-analysis]: Mark A. Weaver (2009). An Interim Analysis Example. Retrieved from [http://www.icssc.org/Documents/AdvBiosGoa/Tab%2025.00_InterimAnalysis.pdf](http://www.icssc.org/Documents/AdvBiosGoa/Tab%2025.00_InterimAnalysis.pdf)
+
+
+One method is to distribute the $$\alpha$$ according to the fraction of information gathered at the point of that analysis[^interim-monitoring]. This fraction of total statistical information gathered is referred to in the literature as the information fraction, and for normally distributed data this information fraction is equal to the proportion of data gathered, $$\frac{N_k}{N_{Sequential}}$$[^interim-monitoring]. Following this, we can choose values of $$\alpha$$ according to: 
+
+$$\alpha_1=\alpha_T\frac{N_1}{N_{Sequential}}^p$$,
+{: style="text-align: center"}
+
+$$\alpha_2-\alpha_1 =\alpha_T\frac{N_2}{N_{Sequential}}^p$$.
```
[the remark this answers is not in the document]
Comment (ngwinglon) [reply _posts/2018-10-12-how-to-experiment-rapidly-without-losing-rigour.md]: I was about to ask the same.

Following the cited Pampallona et al., it seems the rule will be slightly different for the last checkpoint in which case you have to write an equation array in LaTeX for the start, "middle" and final checkpoints?

File: _posts/2018-10-12-how-to-experiment-rapidly-without-losing-rigour.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+1. Under the null hypothesis the probability of the test statistic $$S_1$$ being greater than or equal to the value of the upper boundary at the first checkpoint must be equal to the alpha spending function at the first checkpoint:
+$$P_{H_0}(S_1 \geq u_1) = \alpha_T\frac{N_1}{N_{Sequential}}^p$$.
+2. Under the alternative hypothesis in the case where the true difference between groups is exactly the minimum detectable difference, the probability of the test statistic being less than the lower boundary at the first checkpoint must equal the beta spending function at the first checkpoint:
+$$P_{H_1}(S_1 \leq l_1) = \beta_T\frac{N_1}{N_{Sequential}}^p$$.
+3. Under the null hypothesis the probability that the test statistic falls between the two boundaries $$l_1 \leq S_1 \leq u_1$$ at all previous boundaries and then falls above the upper boundary $$u_k$$ at the $$k$$-th analysis must be equal to difference between the alpha spending function at the $$k$$-th boundary and the spending function at the previous checkpoint:
+$$P_{H_0}(l_1 \leq S_1 \leq u_1, \ldots , l_{k-1} \leq S_{k-1} \leq u_{k-1},S_k\geq u_k) =
```
Comment (ngwinglon) [inline _posts/2018-10-12-how-to-experiment-rapidly-without-losing-rigour.md]: Out of curiosity (because I am not familiar with the literature but interested), and possibly out of scope for this PR:

Why is this a joint probability but not a conditional probability `P_{H_0}(S_k\geq u_k | "have not stopped experiment yet")`?  Or are they the same in this instance?

File: _posts/2018-10-12-how-to-experiment-rapidly-without-losing-rigour.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+$$\sum_{i=1}^k N_i^{Sequential}=N_{Sequential}$$
+	
+* How do we want to distribute (also called “spend” or “split”) the type I and type II errors throughout the experiment, i.e. select $$\alpha_1,\alpha_2,\dots,\alpha_k$$ and $$\beta_1,\beta_2,\ldots,\beta_k$$ corresponding to the above checkpoints.
+
+From these $$\alpha_i$$ and $$\beta_i$$ it is possible to create boundary values for the [test statistic](https://en.wikipedia.org/wiki/Test_statistic) $$S_k$$ at each checkpoint such that if it falls outside of a certain value, we can stop the test and declare a winner. 
+
+This image shows a two sided sequential test design where we have found upper $$u_k$$ and lower $$l_k$$ boundaries for the $$Z$$ test statistic $$Z_k$$ at each checkpoint from a set of $$\alpha_k$$, $$\beta_k$$. If the test statistic falls outside of these boundaries at any checkpoint then we can stop the test early. Derivation of these boundaries is covered below. 
+
+<figure>
+![Example test design](/images/posts/how-to-experiment-rapidly-without-losing-rigour/image_2.png)
```
Comment (ngwinglon) [inline _posts/2018-10-12-how-to-experiment-rapidly-without-losing-rigour.md]: In this plot, the `-`sign of the negative numbers is not clearly visible because of the positioning relative to dashed line. Context is clear though, perhaps it does not need to be changed. Just wanted to flag ...

File: _posts/2018-10-12-how-to-experiment-rapidly-without-losing-rigour.md [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+## Sequential testing
+
+The sequential testing procedure takes the concept of a family-wise error rate[^family-wise-error] and applies it to multiple hypothesis tests across the duration of the experiment. By adjusting the threshold value $$\alpha$$ at which we will reject the null hypothesis at each of the test’s checkpoints, we can maintain the total (i.e. family-wise) probability of making one or more type I errors at a desired level, $$\alpha_{Total}$$.
+
+[^family-wise-error]: Wikipedia contributors. (2018, August 22). Family-wise error rate. In Wikipedia, The Free Encyclopedia. Retrieved 15:43, September 25, 2018, from [https://en.wikipedia.org/w/index.php?title=Family-wise_error_rate&oldid=856111993](https://en.wikipedia.org/w/index.php?title=Family-wise_error_rate&oldid=856111993)
+
+The trade-off is that, by spreading the errors across multiple checkpoints, we need to increase the maximum sample size for the sequential test from $$N_{Fixed}$$ to $$N_{Sequential} > N_{Fixed}$$ in order to maintain $$\beta$$.
```
Comment (ngwinglon) [inline _posts/2018-10-12-how-to-experiment-rapidly-without-losing-rigour.md]: Perhaps delete: " ... _from $$N_{Fixed}$$_ ... " in " ... _from $$N_{Fixed}$$ to $$N_{Sequential} > N_{Fixed}$$_ ... "?

It sounds a bit tautological if one reads "_we need to increase the maximum sample size for the sequential test from $$N_{Fixed}$$ to $$N_{Sequential} > N_{Fixed}$$_"

(as in : "Can you **increase** from x to y such that x <= y?")

File: _posts/2018-10-12-how-to-experiment-rapidly-without-losing-rigour.md [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+
+In the case that our $$\delta_{detectable}$$ estimate is very close to the true value of $$\delta$$, the sequential design would increase the sample size required for our tests from  $$N_{Fixed}$$ to $$N_{Sequential}$$. In many cases though we don’t have a good a priori estimate of the value of $$\delta$$, so to make running the test worthwhile we often err on the side of being able to detect a slightly smaller effect than it may actually have, and thus running the test a bit longer. By introducing checkpoints we are able to set  $$\delta_{detectable}$$ to be quite small, but if the true value of $$\delta$$ is larger we are likely to be able to stop running our test early, i.e. with a sample size less than $$N_{Sequential}$$ (and likely less than $$N_{fixed}$$).
```
Comment (ngwinglon) [inline _posts/2018-10-12-how-to-experiment-rapidly-without-losing-rigour.md]: First sentence sounds a bit repetitive:

"_... increase the sample size required for our tests from  $$N_{Fixed}$$ to $$N_{Sequential}$$_" was just mentioned in the preceding sentence.

File: _posts/2018-10-12-how-to-experiment-rapidly-without-losing-rigour.md (hunk 4 above)
Comment (MahanaC) [reply _posts/2018-10-12-how-to-experiment-rapidly-without-losing-rigour.md]: Thanks, fixed

File: _posts/2018-10-12-how-to-experiment-rapidly-without-losing-rigour.md [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+
+```
+
+<figure>
+![R gsDesign output](/images/posts/how-to-experiment-rapidly-without-losing-rigour/image_4.png)
+</figure>
+
+For a given sequential design the sample size required at the final checkpoint is a multiple of the fixed sample size design. The increase in the maximum sample size needed for this sequential design compared to the fixed design in this case is 
+
+$$N_{Sequential}=1.064 \cdot N_{Fixed}$$.
```
[the remark this answers is not in the document]
Comment (harrysalmon) [reply _posts/2018-10-12-how-to-experiment-rapidly-without-losing-rigour.md]: I think this is ok. The image references the count 1.064 * N_fixed where N_fixed=1000
